cmd/reviewdog/main.go

```diff
@@ -317,23 +317,67 @@ func githubService(ctx context.Context) (gs *githubservice.GitHubPullRequest, is
 	if err != nil {
 		return nil, isPR, err
 	}
-	// TODO: support commit build
-	if !isPR {
-		return nil, isPR, nil
-	}
 
 	client, err := githubClient(ctx, token)
 	if err != nil {
 		return nil, isPR, err
 	}
 
+	if !isPR {
+		pullRequests, _, err := client.PullRequests.List(ctx, g.Owner, g.Repo, &github.PullRequestListOptions{})
+		if err != nil {
+			return nil, false, err
+		}
+
+		if g.Branch != "" {
+			g, isPR, err = getPullRequestIDByBranch(pullRequests, g)
+			if err != nil {
+				return nil, isPR, err
+			}
+		} else if g.SHA != "" {
+			g, isPR, err = getPullRequestIDByCommit(pullRequests, g)
+			if err != nil {
+				return nil, isPR, err
+			}
+		}
+	}
+
+	//// TODO: support commit build
+	if !isPR {
+		return nil, isPR, nil
+	}
+
 	gs, err = githubservice.NewGitHubPullRequest(client, g.Owner, g.Repo, g.PullRequest, g.SHA)
 	if err != nil {
 		return nil, isPR, err
 	}
 	return gs, isPR, nil
 }
 
+func getPullRequestIDByBranch(pullRequests []*github.PullRequest, info *cienv.BuildInfo) (*cienv.BuildInfo, bool, error) {
```

Comment:
Pass `pullRequests` and branch and return pr number and error.

You can change isPR based on error status and log error in caller side.
You can refer fetchMergeRequestIDFromCommit function.